jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenSettingsServerCredentials.java

```diff
@@ -32,31 +38,50 @@
   @VisibleForTesting static final String CREDENTIAL_SOURCE = "Maven settings";
 
   private final Settings settings;
+  private final SettingsDecrypter settingsDecrypter;
 
-  MavenSettingsServerCredentials(Settings settings) {
+  MavenSettingsServerCredentials(Settings settings, @Nullable SettingsDecrypter settingsDecrypter) {
     this.settings = settings;
+    this.settingsDecrypter = settingsDecrypter;
   }
 
   /**
    * Attempts to retrieve credentials for {@code registry} from Maven settings.
    *
    * @param registry the registry
    * @return the credentials for the registry
+   * @throws MojoExecutionException if the credentials could not be retrieved
    */
   @Nullable
-  RegistryCredentials retrieve(@Nullable String registry) {
+  RegistryCredentials retrieve(@Nullable String registry) throws MojoExecutionException {
     if (registry == null) {
       return null;
     }
 
-    Server registryServerSettings = settings.getServer(registry);
-    if (registryServerSettings == null) {
+    Server registryServer = settings.getServer(registry);
+    if (registryServer == null) {
       return null;
     }
 
+    if (settingsDecrypter != null) {
+      SettingsDecryptionRequest request = new DefaultSettingsDecryptionRequest(registryServer);
+      SettingsDecryptionResult result = settingsDecrypter.decrypt(request);
+      // un-encrypted passwords are passed through, so a problem indicates a real issue
+      for (SettingsProblem problem : result.getProblems()) {
+        if (problem.getSeverity() == SettingsProblem.Severity.ERROR
+            || problem.getSeverity() == SettingsProblem.Severity.FATAL) {
+          throw new MojoExecutionException(
+              "Unable to decrypt settings for " + registry + ": " + problem);
```

Comment:
I think "... decrypt password for ..." is more informative to actual users and has the potential to make them actions on their own.